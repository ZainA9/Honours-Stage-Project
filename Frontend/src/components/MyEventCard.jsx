import React from 'react';

const DEFAULT_IMAGE   = '/images/event-fallback.jpg';
const CATEGORY_IMAGES = {
  Technology: '/images/technology.jpg',
  Music:      '/images/music.jpg',
  Food:       '/images/food.jpg',
  Arts:       '/images/arts.jpg',
  Sports:     '/images/sports.jpg',
  Education:  '/images/education.jpg',
};

export default function MyEventCard({ event, onEdit, onDelete }) {
  const { id, name, location, date, categories, imageUrl } = event;
  const category = categories?.[0] || 'General';

  // event.imageUrl ‚Üí local category image ‚Üí fallback
  const src =
    imageUrl ||
    CATEGORY_IMAGES[category] ||
    DEFAULT_IMAGE;

  return (
    <div className="bg-white rounded-xl shadow overflow-hidden">
      <img
        src={src}
        alt={name}
        className="h-52 w-full object-cover"
        onError={e => {
          e.currentTarget.onerror = null; 
          e.currentTarget.src = DEFAULT_IMAGE;
        }}
      />
      <div className="p-4">
        <span className="inline-block text-sm bg-indigo-100 text-indigo-600 px-2 py-0.5 rounded-full mb-2">
          {category}
        </span>
        <h4 className="font-semibold text-lg">{name}</h4>
        <p className="text-sm text-gray-500 my-2">
          üìÖ {new Date(date).toLocaleDateString()}
        </p>
        <p className="text-sm text-gray-500 mb-4">üìç {location}</p>
        <div className="flex gap-2">
          <button
            onClick={() => onEdit(id)}
            className="flex-1 bg-blue-600 text-white px-4 py-2 rounded hover:bg-blue-700"
          >
            Edit Event
          </button>
          <button
            onClick={() => onDelete(id)}
            className="flex-1 bg-red-600 text-white px-4 py-2 rounded hover:bg-red-700"
          >
            Delete Event
          </button>
        </div>
      </div>
    </div>
  );
}
