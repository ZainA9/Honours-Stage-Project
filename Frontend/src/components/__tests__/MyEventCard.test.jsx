import React from 'react';
import { render } from '@testing-library/react';
import MyEventCard from '../MyEventCard';

describe('MyEventCard', () => {
  const mockEvent = {
    id: '1',
    name: 'Test Event',
    location: 'Test City',
    date: '2024-05-20',
    categories: ['Tech'],
    imageUrl: null,
  };

  const noop = () => {};

  it('renders correctly', () => {
    const { asFragment } = render(
      <MyEventCard event={mockEvent} onEdit={noop} onDelete={noop} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
