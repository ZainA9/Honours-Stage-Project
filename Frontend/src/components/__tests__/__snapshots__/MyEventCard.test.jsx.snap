// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyEventCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="bg-white rounded-xl shadow overflow-hidden"
  >
    <img
      alt="Test Event"
      class="h-52 w-full object-cover"
      src="/images/event-fallback.jpg"
    />
    <div
      class="p-4"
    >
      <span
        class="inline-block text-sm bg-indigo-100 text-indigo-600 px-2 py-0.5 rounded-full mb-2"
      >
        Tech
      </span>
      <h4
        class="font-semibold text-lg"
      >
        Test Event
      </h4>
      <p
        class="text-sm text-gray-500 my-2"
      >
        📅 20/05/2024
      </p>
      <p
        class="text-sm text-gray-500 mb-4"
      >
        📍 Test City
      </p>
      <div
        class="flex gap-2"
      >
        <button
          class="flex-1 bg-blue-600 text-white px-4 py-2 rounded hover:bg-blue-700"
        >
          Edit Event
        </button>
        <button
          class="flex-1 bg-red-600 text-white px-4 py-2 rounded hover:bg-red-700"
        >
          Delete Event
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
